### PR TITLE
Update proot

### PR DESCRIPTION
**Pull request by corbinlc**
## What changes does this PR introduce?
Take another crack at fixing chdir issues, but also take a crack at fixing stat64 issues.
Also, determine which arch was actually unpacked for the libraries

## Any background context you want to provide?
Just trying to work through some current users issues

## Where should the reviewer start?
UlaFiles.kt

## Has this been manually tested? How?
Yes, I ran in debug and made sure it listed out the expected arch on a couple of devices.

## What value does this provide to our end users?
They will be able to actually use our app if they are affected by these issues,

## What GIF best describes this PR or how it makes you feel?
I miss you guys

### CODE REVIEW

File: app/src/main/java/tech/ula/model/repositories/AssetRepository.kt [hunk 1]
```diff
@@ -24,7 +24,8 @@ data class DownloadMetadata(
 class AssetRepository(
     private val applicationFilesDirPath: String,
     private val assetPreferences: AssetPreferences,
-    private val githubApiClient: GithubApiClient = GithubApiClient(),
+    private val ulaFiles: UlaFiles,
+    private val githubApiClient: GithubApiClient = GithubApiClient(ulaFiles),
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/model/repositories/AssetRepository.kt]: I think the initialization of `GithubApiClient` should probably be handled by whatever is constructing the `AssetRepository`, in order to match code style in the rest of the app. That is to say, we shouldn't use a default parameter here.

File: app/src/main/java/tech/ula/ui/FilesystemEditFragment.kt [hunk 2]
```diff
@@ -233,7 +233,8 @@ class FilesystemEditFragment : Fragment() {
             navController.popBackStack()
         } else {
             try {
-                filesystem.archType = DeviceArchitecture().getArchType()
+                val ulaFiles = UlaFiles(activityContext, activityContext.applicationInfo.nativeLibraryDir)
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/ui/FilesystemEditFragment.kt]: It doesn't look like this needs to be surrounded in a `try catch` block anymore.

**Comment by corbinlc** [comment]
Hey @MatthewTighe, report_coverage is failing every time.  Is this something I broke?

File: app/src/main/java/tech/ula/ui/FilesystemEditFragment.kt (hunk 2 above)
Comment (corbinlc) [reply app/src/main/java/tech/ula/ui/FilesystemEditFragment.kt]: I removed the `try/catch`

File: app/src/main/java/tech/ula/model/repositories/AssetRepository.kt (hunk 1 above)
Comment (corbinlc) [reply app/src/main/java/tech/ula/model/repositories/AssetRepository.kt]: True.  This has been made to follow the style.

**Comment by MatthewTighe** [comment]
No, the coverage reporting issue doesn't look like anything you did. I'd have to look into fixing it.

**Pull request merged**